File: Baekjoon/2636.py
from collections import deque
import sys
input = sys.stdin.readline 

N, M = map(int, input().split())
graph = []
visited = [[False for _ in range(M)]for _ in range(N)]
for i in range(N):
    graph.append(list(map(int, input().split())))

ans = []
dx = [1, -1,0,0]
dy = [0,0,1,-1]
def bfs():
    queue = deque()
    queue.append([0,0])
    visited[0][0] = True 
    cnt = 0
    while queue:
        x,y = queue.popleft()
        for num in range(4):
            xx = x + dx[num]
            yy = y + dy[num]
            if 0 <= xx < N and 0 <= yy < M and not visited[xx][yy]:
                if graph[xx][yy] == 0:
                    visited[xx][yy] = True 
                    queue.append([xx,yy])
                elif graph[xx][yy] == 1:
                    graph[xx][yy] = 0
                    visited[xx][yy] = True 
                    cnt += 1
    ans.append(cnt)
    return cnt

time = 0

while True:
    time +=1
    visited = [[False]*M for _ in range(N)]
    cnt = bfs() 
    if cnt == 0:
        break
print(time-1)
print(ans[-2])

File: JavaScript/Baekjoon/2636.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const board = input.slice(1, n + 1).map((line) => line.split(" ").map(Number));

const dx = [1, -1, 0, 0];
const dy = [0, 0, 1, -1];
let ans = [];

function bfs() {
  const queue = [[0, 0]];
  const visited = Array.from({ length: n }, () => Array(m).fill(0));
  visited[0][0] = 1;
  let count = 0;

  while (queue.length > 0) {
    const [x, y] = queue.shift();
    for (let i = 0; i < 4; i++) {
      const nx = x + dx[i];
      const ny = y + dy[i];
      if (nx >= 0 && nx < n && ny >= 0 && ny < m && visited[nx][ny] === 0) {
        if (board[nx][ny] === 0) {
          visited[nx][ny] = 1;
          queue.push([nx, ny]);
        } else if (board[nx][ny] === 1) {
          board[nx][ny] = 0;
          visited[nx][ny] = 1;
          count++;
        }
      }
    }
  }
  ans.push(count);
  return count;
}

let time = 0;
while (true) {
  time += 1;
  const cnt = bfs();
  if (cnt === 0) {
    break;
  }
}

console.log(time - 1);
console.log(ans[ans.length - 2]); // Accessing the second last element
